src/main/java/com/answerdigital/answerking/repository/CategoryRepository.java

```diff
@@ -13,4 +14,6 @@ public interface CategoryRepository extends PagingAndSortingRepository<Category,
     boolean existsByNameAndIdIsNot(String name, Long id);
 
     Set<Category> findAll();
+
+    List<Category> findCategoriesByProductsId(Long productId);
```

Comment:
I think this should be singular as it finds by single product id.: 
`List<Category> findCategoriesByProductId(Long productId);`

Reply:
made the changes.